### PR TITLE
Refactor of facets

### PR DESCRIPTION
**Pull request by koetsier**
- Set the value as part of the initializer in Facets instead of through setter
- Handle setting value of facets in subclasses of Facets because they mean different things in those subclasses.
- Set the values in FacetCollection as part of the initializer
- Remove unused code in FacetTagPresenter and simplify code

### CODE REVIEW

File: app/presenters/finder_presenter.rb [hunk 1]
```diff
@@ -345,4 +334,15 @@ def alert_query_string
   def eu_exit_finder
     slug == "/find-eu-exit-guidance-business"
   end
+
+  def facet_hashes(content_item_hash)
+    FacetExtractor.for(content_item_hash).extract
```
Comment (bilbof) [inline app/presenters/finder_presenter.rb]: What is a `FacetExtractor`? Forgive my ignorance, it looks this was already in the codebase, but I'm not sure what it is.

File: app/models/hidden_facet.rb [hunk 2]
```diff
@@ -1,4 +1,11 @@
 class HiddenFacet < FilterableFacet
+  attr_reader :value
+
+  def initialize(facet, value)
```
Comment (oscarwyatt) [inline app/models/hidden_facet.rb]: I heartily approve of this

File: app/models/filterable_facet.rb [hunk 3]
```diff
@@ -1,8 +1,10 @@
 class FilterableFacet < Facet
-  attr_accessor :value
-
   DEFAULT_PREPOSITION = 'related to'.freeze
 
+  def initialize(facet)
+    super(facet)
+  end
```
[the remark this answers is not in the document]
Comment (koetsier) [reply app/models/filterable_facet.rb]: removed

File: app/models/taxon_facet.rb [hunk 4]
```diff
@@ -1,7 +1,26 @@
...
+class TaxonFacet < FilterableFacet
   LEVEL_ONE_TAXON_KEY = 'level_one_taxon'.freeze
   LEVEL_TWO_TAXON_KEY = 'level_two_taxon'.freeze
 
+  def initialize(facet, value_hash)
+    @value_hash = value_hash
+    super(facet)
+  end
+
+  def options
```
[the remark this answers is not in the document]
Comment (koetsier) [reply app/models/taxon_facet.rb]: removed

**Pull request merged**